core-bundle/src/DependencyInjection/ContaoCoreExtension.php

```diff
@@ -63,7 +65,7 @@ public function load(array $configs, ContainerBuilder $container): void
         $loader->load('services.yml');
         $loader->load('migrations.yml');
 
-        $container->setParameter('contao.web_dir', $config['web_dir']);
+        $container->setParameter('contao.web_dir', $this->getComposerPublicDir($projectDir) ?? $config['web_dir']);
```

Comment:
This is a forward compatible implementation. It should allow us to remove the `contao.web_dir` config entry in Contao 5 and simply make `public` the fallback here:
```suggestion
        $container->setParameter('contao.web_dir', $this->getComposerPublicDir($projectDir) ?? 'public');
```

Reply:
You mean `Path::join($projectDir, 'public')`, don't you?